File: packages/sdk/go/internal/model/videos.go
// Package model 的视频(生成 / 任务查询 / 编排)类型。
package model

// VideoGenerationsRequest 是 POST /v1/videos/generations 请求体(视频生成)。
type VideoGenerationsRequest struct {
	Model      string `json:"model"`
	Prompt     string `json:"prompt"`
	Image      string `json:"image,omitempty"`
	Duration   *int   `json:"duration,omitempty"`
	Resolution string `json:"resolution,omitempty"`
	Vendor     string `json:"vendor,omitempty"`
}

// VideoGenerationsResponse 是 POST /v1/videos/generations 响应体(异步任务)。
type VideoGenerationsResponse struct {
	TaskID        string `json:"taskId"`
	Status        string `json:"status"`
	EstimatedTime *int   `json:"estimatedTime,omitempty"`
}

// VideoTaskResponse 是 GET /v1/videos/tasks/:id 响应体。
type VideoTaskResponse struct {
	TaskID    string  `json:"taskId"`
	Status    string  `json:"status"`
	VideoURL  string  `json:"videoUrl,omitempty"`
	Progress  float64 `json:"progress,omitempty"`
	Error     string  `json:"error,omitempty"`
	CreatedAt string  `json:"createdAt"`
}

// VideoComposeScene 是视频编排分镜。
type VideoComposeScene struct {
	Text        string `json:"text"`
	Duration    int    `json:"duration"`
	ImagePrompt string `json:"imagePrompt,omitempty"`
}

// VideoComposeRequest 是 POST /v1/videos/compose 请求体(视频编排)。
type VideoComposeRequest struct {
	Scenes []VideoComposeScene `json:"scenes"`
	BgmURL string              `json:"bgmUrl,omitempty"`
}

// VideoComposeResponse 是 POST /v1/videos/compose 响应体。
type VideoComposeResponse struct {
	ComposeID string `json:"composeId"`
	Status    string `json:"status"`
}

File: sdks/typescript/src/videos.ts
/**
 * 视频模块 — 生成 / 任务查询 / 编排。
 *
 * 端点(3 个):
 * - POST /v1/videos/generations
 * - GET  /v1/videos/tasks/:id
 * - POST /v1/videos/compose
 */

import type { BaseClient } from './base.js'

// =============================================================================
// 内联类型定义
// =============================================================================

export interface V1VideoGenerationsRequest {
  model: string
  prompt: string
  image?: string
  duration?: number
  resolution?: string
  vendor?: string
}

export interface V1VideoGenerationsResponse {
  taskId: string
  status: 'pending' | 'processing' | 'completed' | 'failed'
  estimatedTime?: number
}

export interface V1VideoTaskResponse {
  taskId: string
  status: 'pending' | 'processing' | 'completed' | 'failed'
  videoUrl?: string
  progress?: number
  error?: string
  createdAt: string
}

export interface V1VideoComposeRequest {
  scenes: Array<{
    text: string
    duration: number
    imagePrompt?: string
  }>
  bgmUrl?: string
}

export interface V1VideoComposeResponse {
  composeId: string
  status: 'processing' | 'completed' | 'failed'
}

// =============================================================================
// Module 接口 + 工厂函数
// =============================================================================

export interface VideosModule {
  /** POST /v1/videos/generations(视频生成,异步任务)。 */
  generations(req: V1VideoGenerationsRequest): Promise<V1VideoGenerationsResponse>
  /** GET /v1/videos/tasks/:id(查询视频任务状态)。 */
  getTask(taskId: string): Promise<V1VideoTaskResponse>
  /** POST /v1/videos/compose(视频编排)。 */
  compose(req: V1VideoComposeRequest): Promise<V1VideoComposeResponse>
}

export function createVideosModule(client: BaseClient): VideosModule {
  return {
    generations: (req) =>
      client.request<V1VideoGenerationsResponse>('POST', '/videos/generations', req),
    getTask: (taskId) =>
      client.request<V1VideoTaskResponse>('GET', `/videos/tasks/${encodeURIComponent(taskId)}`),
    compose: (req) => client.request<V1VideoComposeResponse>('POST', '/videos/compose', req),
  }
}
